controllers/account.go

```diff
@@ -236,22 +237,47 @@ func (c *ApiController) Signup() {
 
 // Logout
 // @Title Logout
-// @Tag Login API
+// @Tag Logout API
 // @Description logout the current user
+// @Param   id_token_hint   query        string  false        "id_token_hint"
+// @Param   post_logout_redirect_uri    query    string  false     "post_logout_redirect_uri"
+// @Param   state     query    string  false     "state"
 // @Success 200 {object} controllers.Response The Response object
 // @router /logout [get,post]
 func (c *ApiController) Logout() {
 	user := c.GetSessionUsername()
 	util.LogInfo(c.Ctx, "API: [%s] logged out", user)
 
-	application := c.GetSessionApplication()
-	c.ClearUserSession()
+	// https://openid.net/specs/openid-connect-rpinitiated-1_0-final.html
+	accessToken := c.Input().Get("id_token_hint")
+	redirectUri := c.Input().Get("post_logout_redirect_uri")
+	state := c.Input().Get("state")
 
-	if application == nil || application.Name == "app-built-in" || application.HomepageUrl == "" {
-		c.ResponseOk(user)
-		return
+	if accessToken == "" {
+		application := c.GetSessionApplication()
+		c.ClearUserSession()
+
+		if application == nil || application.Name == "app-built-in" || application.HomepageUrl == "" {
+			c.ResponseOk(user)
+			return
+		}
+		c.ResponseOk(user, application.HomepageUrl)
+	}
+
+	validate, application := object.DeleteTokenByAccessToken(accessToken)
```

Comment:
The token table is more like logging, it should not be deleted. Maybe we want to use session table added by PR: https://github.com/casdoor/casdoor/pull/1441 ?